lib/strategies/base.js

```diff
@@ -53,6 +53,10 @@ BaseStrategy.prototype._attachListeners = function () {
     self._reformatInput(event);
   });
   self.inputElement.addEventListener('input', function (event) {
+    // Safari AutoFill fires CustomEvents -- mark the input as unformatted without actually unformatting the current value
+    if (!(event instanceof KeyboardEvent)) {
```

Comment:
As-is, it makes an implicit assumption that if something is firing the input listener that isn't a keyboard event, the input value is "dirty" because the formatter hasn't formatted the value yet. I think it makes sense to make it more explicit to avoid the scope of the fix possibly affecting the _unknown-unknown_ cases.